tools/clang/unittests/HLSLTestLib/FileCheckerTest.cpp

```diff
@@ -1091,6 +1103,59 @@ FileRunCommandResult FileRunCommandPart::RunDxilVer(dxc::DxcDllSupport& DllSuppo
   return CheckDxilVer(DllSupport, RequiredDxilMajor, RequiredDxilMinor);
 }
 
+#ifdef ENABLE_DXIL2SPV
+FileRunCommandResult
+FileRunCommandPart::RunDxil2Spv(dxc::DxcDllSupport &DllSupport,
+                                const FileRunCommandResult *Prior) {
+  // Support piping stdin from prior if needed.
+  UNREFERENCED_PARAMETER(Prior);
+  bool success = true;
+
+  std::string stdoutString;
+  std::string stderrString;
+  llvm::raw_string_ostream stdoutStream(stdoutString);
+  llvm::raw_string_ostream stderrStream(stderrString);
+
+  try {
+    // Configure filesystem.
+    llvm::sys::fs::MSFileSystem *msf;
+    HRESULT hr;
+    if (SUCCEEDED(hr = CreateMSFileSystemForDisk(&msf))) {
+      llvm::sys::fs::AutoPerThreadSystem pts(msf);
+      IFTLLVM(pts.error_code());
+
+      // Set up diagnostics.
+      clang::CompilerInstance instance;
+      auto *diagnosticPrinter = new clang::TextDiagnosticPrinter(
+          stderrStream, new clang::DiagnosticOptions());
+      instance.createDiagnostics(diagnosticPrinter, false);
+      instance.setOutStream(&stdoutStream);
+
+      // Set compiler options.
+      instance.getCodeGenOpts().MainFileName = CW2A(CommandFileName, CP_UTF8);
+      instance.getCodeGenOpts().SpirvOptions.targetEnv = "vulkan1.0";
+
+      clang::dxil2spv::Translator translator(instance);
+      translator.Run();
+    } else {
+      success = false;
+    }
+  } catch (...) {
+    success = false;
+  }
+
+  stdoutStream.flush();
+  stderrStream.flush();
+
+  FileRunCommandResult result = {};
+  result.StdOut = stdoutString;
+  result.StdErr = stderrString;
```

Comment:
nit: You can do `stdoutStream.str()` and it will `flush` automatically for you

Reply:
Done